pgraph/actions.go

```diff
@@ -614,6 +614,7 @@ func (g *Graph) Worker(v *Vertex) error {
 func (g *Graph) Start(first bool) { // start or continue
 	log.Printf("State: %v -> %v", g.setState(graphStateStarting), g.getState())
 	defer log.Printf("State: %v -> %v", g.setState(graphStateStarted), g.getState())
+	defer g.prometheus.UpdatePgraphStartTime()
```

Comment:
Q: Do you really want defer, and not just run at the end of the function?
Also, FYI: Start() gets called for Start and Unpause just so you know.